src/applications/Machine.java

```diff
@@ -15,39 +15,57 @@
         jobQ = new LinkedQueue();
     }
 
-    public LinkedQueue getJobQ() {
-        return jobQ;
-    }
-
-    public int getChangeTime() {
-        return changeTime;
-    }
+    /**
+     * change the state of theMachine
+     *
+     * @return last job run on this machine
+     */
+    Job changeState(Machine[] machine, EventList eList, int largeTime, int timeNow, int theMachine) {// Task on theMachine has finished,
```

Comment:
I feel like this is a lot of stuff to be passing to a method. It this smells of long parameter length.

Reply:
That is what I also thought as well for being an issue. I'm not sure how to reduce the number of parameters, so I just moved the method back to machineshopsimulator.java